plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/config/LibraryBuilder.java

```diff
@@ -0,0 +1,107 @@
+package com.google.cloud.tools.eclipse.appengine.libraries.config;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.LinkedList;
+import java.util.List;
+
+import org.eclipse.core.runtime.IConfigurationElement;
+import org.eclipse.core.runtime.InvalidRegistryObjectException;
+
+import com.google.cloud.tools.eclipse.appengine.libraries.Filter;
+import com.google.cloud.tools.eclipse.appengine.libraries.Library;
+import com.google.cloud.tools.eclipse.appengine.libraries.LibraryFile;
+import com.google.cloud.tools.eclipse.appengine.libraries.MavenCoordinates;
+
+public class LibraryBuilder {
+
+  private static final String LIBRARY = "library";
+  private static final String ID = "id";
+  private static final String NAME = "name";
+  private static final String SITE_URI = "siteUri";
+  private static final String LIBRARY_FILE = "libraryFile";
+  private static final String EXCLUSION_FILTER = "exclusionFilter";
+  private static final String INCLUSION_FILTER = "inclusionFilter";
+  private static final String MAVEN_COORDINATES = "mavenCoordinates";
+  private static final String SOURCE_URI = "sourceUri";
+  private static final String JAVADOC_URI = "javadocUri";
+  private static final String PATTERN = "pattern";
+  private static final String GROUP_ID = "groupId";
+  private static final String REPOSITORY_URI = "repositoryUri";
+  private static final String ARTIFACT_ID = "artifactId";
+  private static final String VERSION = "version";
+  private static final String TYPE = "type";
+  private static final String CLASSIFIER = "classifier";
+
+  public Library build(IConfigurationElement configurationElement) {
+    try {
+    if (configurationElement.getName().equals(LIBRARY)) {
+      Library library = new Library(configurationElement.getAttribute(ID));
+      library.setName(configurationElement.getAttribute(NAME));
+      library.setSiteUri(new URI(configurationElement.getAttribute(SITE_URI)));
+      library.setLibraryFiles(getLibraryFiles(configurationElement.getChildren(LIBRARY_FILE)));
+      return library;
+    }
+    } catch (InvalidRegistryObjectException | URISyntaxException e) {
```

Comment:
Although your schemas may say attributes are required, the user may have used a text file and gotten things wrong.  So you may have NPEs and/or `IllegalArgumentException`s (raised by `Preconditions` checks) that should either be caught here.  Or you should have `null` checks in the rest of this code.